src/azure-cli-core/azure/cli/core/commands/cache.py

```diff
@@ -0,0 +1,131 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import os
+import timeit
+
+from knack.log import get_logger
+from knack.config import get_config_parser
+
+from azure.cli.core.extension import get_extensions
+from azure.cli.core._environment import get_config_dir
+
+logger = get_logger(__name__)
+
+
+CACHE_CONFIG_FILE = os.path.join(get_config_dir(), 'command_module_index.config')
+
+VERSION_STR = "VERSION"
+EXTENSIONS_STR = "EXTENSIONS"
+EXTENSIONS_LIST = ["{}-{}".format(ext.name, ext.version) for ext in get_extensions()]
+INDEX_STR = "INDEX"
+
+
+# return true if the cache exists and caching is enabled.
+def cache_exists():
+    use_cache = os.path.isfile(CACHE_CONFIG_FILE) and os.getenv('AZ_USE_CACHE', 'False').lower() == 'true'
+    logger.debug('use_cache is %s', use_cache)
+    return use_cache
+
+
+def cache_command_table(cli_ctx, cmd_to_loader_map):
+
+    config = get_config_parser()
+    config.read(CACHE_CONFIG_FILE)
+
+    config.remove_section(VERSION_STR)
+    config.remove_section(EXTENSIONS_STR)
+    config.remove_section(INDEX_STR)
+
+    config.add_section(VERSION_STR)
+    config.add_section(EXTENSIONS_STR)
+    config.add_section(INDEX_STR)
```

Comment:
This is odd. It would be simpler to simply overwrite the file entirely. JSON would make this easy.